src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs

```diff
@@ -139,6 +167,47 @@ private static void InstrumentRequest(HttpWebRequest request, Activity activity)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsRequestInstrumented(HttpWebRequest request)
+            => request.Headers.Get(TraceParentHeaderName) != null;
+
+        private static void ProcessRequest(HttpWebRequest request)
+        {
+            if (!WebRequestActivitySource.HasListeners() || IsRequestInstrumented(request))
```

Comment:
the check for `.HasListeners`() is just trying to optimize right? Its not actually required?

Reply:
Correct